### PR TITLE
Commas trailing

### PR DESCRIPTION
**Pull request by MEO265**
This is the pull request for [issue #2102](https://github.com/r-lib/lintr/issues/2102)

This is my first PR in your project, please don't hold back with comments

### CODE REVIEW

File: R/commas_linter.R [hunk 1]
```diff
@@ -58,7 +71,11 @@ commas_linter <- function() {
     @line1 = preceding-sibling::*[1]/@line1 and
     not(preceding-sibling::*[1][self::OP-COMMA or self::EQ_SUB])
   ]"
-  xpath_after <- "//OP-COMMA[@line1 = following-sibling::*[1]/@line1 and @col1 = following-sibling::*[1]/@col1 - 1]"
+  xpath_after <- paste0(
+    "//OP-COMMA[@line1 = following-sibling::*[1]/@line1 and @col1 = following-sibling::*[1]/@col1 - 1",
+    if (allow_trailing_comma) " and not(following-sibling::*[1]/self::OP-RIGHT-BRACKET)",
```
Comment (AshesITR) [inline R/commas_linter.R:76]: Maybe also allow `RBB` (`a[[1,]]`) and `OP-RIGHT-PAREN` (`a(1,)`) if the argument is set?

**Review by AshesITR — COMMENTED** [review]
Thanks a lot for the PR. Solid work, just a few remarks.

File: R/commas_linter.R [hunk 2]
```diff
@@ -40,12 +48,17 @@
...
+#'   linters = commas_linter(allow_trailing_comma = TRUE)
+#' )
+#'
 #' @evalRd rd_tags("commas_linter")
 #' @seealso
 #' - [linters] for a complete list of linters available in lintr.
 #' - <https://style.tidyverse.org/syntax.html#commas>
 #' @export
-commas_linter <- function() {
+commas_linter <- function(allow_trailing_comma = FALSE) {
```
[the remark this answers is not in the document]
Comment (MichaelChirico) [reply R/commas_linter.R:61]: it's also more consistent with missing_argument_linter()

**Comment by MEO265** [comment]
[Line 76](https://github.com/r-lib/lintr/pull/2104/commits/2838a6a0cdef36909a3f8daa0943d5241f358e2f#file-r-commas_linter-r-L76) of `commas_linter.R` is 121 characters and not 120 characters long, should that be changed or is there tolerance?
I don't think the whole thing became more readable due to a change

**Comment by AshesITR** [comment]
There is no tolerance, 120 is an upper limit.
Regarding the long line here, it could be broken up by introducing a local variable.

```r
if (allow_trailing) {
  xp_trailing_cond <- "and not(following-sibling::*[1][
    self::OP-RIGHT-PAREN or self::OP-RIGHT-BRACKET or self::RBB
  ]"
} else {
  xp_trailing_cond <- ""
}
xpath_after <- glue(
  "//OP-COMMA[@line1 = following-sibling::*[1]/@line1 and @col1 = following-sibling::*[1]/@col1 - 1 {xp_trailing_cond}]"
)
```

**Comment by MEO265** [comment]
> There is no tolerance, 120 is an upper limit. Regarding the long line here, it could be broken up by introducing a local variable.
> 
> ```r
> if (allow_trailing) {
>   xp_trailing_cond <- "and not(following-sibling::*[1][
>     self::OP-RIGHT-PAREN or self::OP-RIGHT-BRACKET or self::RBB
>   ]"
> } else {
>   xp_trailing_cond <- ""
> }
> xpath_after <- glue(
>   "//OP-COMMA[@line1 = following-sibling::*[1]/@line1 and @col1 = following-sibling::*[1]/@col1 - 1 {xp_trailing_cond}]"
> )
> ```

@AshesITR I have already implemented another solution, should I still implement the one suggested?

**Review by AshesITR — APPROVED** [review]

**Comment by AshesITR** [comment]
It's fine, thanks!

**Pull request merged**